pipeline/mist.go

```diff
@@ -64,30 +62,6 @@ func (m *mist) HandleStartUploadJob(job *JobInfo) (*HandlerOutput, error) {
 	log.AddContext(job.RequestID, "mist_target_url", mistTargetURL)
 	log.AddContext(job.RequestID, "segmented_url", job.SegmentingTargetURL)
 
-	// Arweave / IPFS URLs don't support HTTP Range requests and so Mist can't natively handle them for segmenting
-	// This workaround copies the file from Arweave to S3 and then tells Mist to use the S3 URL
-	if clients.IsDStorageResource(job.SourceFile) {
-		if !isVideo(job.RequestID, job.SourceFile) {
-			return nil, fmt.Errorf("source was not a video: %s", job.SourceFile)
-		}
-		sourceURL, err := url.Parse(job.SourceFile)
-		if err != nil {
-			return nil, fmt.Errorf("failed to parse source as URL: %w", err)
-		}
-
-		newSourceURL, err := inSameDirectory(*sourceOutputUrl, MIST_SEGMENTING_SUBDIR, path.Base(sourceURL.Path))
-		if err != nil {
-			return nil, fmt.Errorf("cannot create location for source copy: %w", err)
-		}
-		log.AddContext(job.RequestID, "new_source_url", newSourceURL.String())
-
-		if err := clients.CopyDStorageToS3(job.SourceFile, newSourceURL.String(), job.RequestID); err != nil {
```

Comment:
So where does the DStorage copy logic live now?

Reply:
Since we're using the unified copy now it gets taken care of there, `input_copy.go` handles reading from dstorage https://github.com/livepeer/catalyst-api/pull/463/files#diff-d93900b04efde86efa555721e97460e1caf9b798bbdae4a68a1f4105c0500160R227